av.c

```diff
@@ -291,24 +288,23 @@ Perl_av_fetch(pTHX_ AV *av, SSize_t key, I32 lval)
         }
     }
 
-    neg  = (key < 0);
-    size = AvFILLp(av) + 1;
-    key += neg * size; /* handle negative index without using branch */
+    /* recalculating key early helps branch prediction */
+    SSize_t neg = key < 0;
+    SSize_t size = AvFILLp(av) + 1;
+    key += neg * size;
 
     /* the cast from SSize_t to Size_t allows both (key < 0) and (key >= size)
-     * to be tested as a single condition */
-    if ((Size_t)key >= (Size_t)size) {
-        if (UNLIKELY(neg))
-            return NULL;
-        goto emptiness;
-    }
-
-    if (!AvARRAY(av)[key]) {
-      emptiness:
-        return lval ? av_store(av,key,newSV_type(SVt_NULL)) : NULL;
-    }
-
-    return &AvARRAY(av)[key];
+     * to be tested as a single condition. */
```

Comment:
I would drop the addition of the trailing period to the comment. None of the new comments added in this change have periods and since the lines before and after this line are unchanged it should make the diff a bit easier to read.

Reply:
done